std/internal/test/randomized_unittest_benchmark.d

```diff
@@ -0,0 +1,495 @@
+}
+
+struct Benchmark
+{
+    string filename; // where to write the benchmark result to
+    string funcname; // the name of the benchmark
+    string timeScale; // the unit the benchmark is measuring in
+    real avgStopWatch; // the avg time it takes to get the clocktime twice
+    size_t rounds; // how many round the benchmark is run
+    bool dontWrite; // if set, no data is written to the the file name "filename"
+    bool wasInterrupted; // true if, RndValueGen opApply was interrupt unexpectitally
+    StopWatch stopWatch; // the time taking device
+    TickDuration[] ticks; // the stopped times, there will be rounds ticks
+    size_t ticksIndex = 0; // the index into ticks
+
+    alias stopWatch this;
+
+    static auto opCall(in string funcname, in size_t rounds,
+            in string filename = __FILE__)
+    {
+        Benchmark ret;
+        ret.filename = filename;
+        ret.funcname = funcname;
+        ret.timeScale = "nsecs";
+        ret.rounds = rounds;
+        ret.ticks = new TickDuration[ret.rounds];
+        ret.avgStopWatch = avgStopWatchTime();
+        return ret;
+    }
+
+    void start()
+    {
+        this.stopWatch.start();
+    }
+
+    void stop()
+    {
+        this.stopWatch.stop();
+        if (this.ticksIndex < ticks.length)
+        {
+            this.ticks[this.ticksIndex] = this.stopWatch.peek();
+            ++this.ticksIndex;
+        }
+    }
+
+    ~this()
+    {
+        import std.stdio : File;
+        if (!this.dontWrite)
+        {
+            import std.algorithm : sort;
+            this.ticks.sort();
+
+            auto f = File(filename ~ "_bechmark.csv", "a");
+            scope(exit) f.close();
+
+            auto q0 = ticks[0].to!("nsecs",real)() / this.rounds;
+            auto q25 = getQuantilTick!0.25(ticks).to!("nsecs",real)() / this.rounds;
+            auto q50 = getQuantilTick!0.50(ticks).to!("nsecs",real)() / this.rounds;
+            auto q75 = getQuantilTick!0.75(ticks).to!("nsecs",real)() / this.rounds;
+            auto q100 = this.ticks[$-1].to!("nsecs",real)() / this.rounds;
+
+            // funcName, the data when the benchmark was created, unit of time,
+            // rounds, wasInterrupted, avgStopWatch, low, 0.25 quantil, median,
+            // 0.75 quantil, high
+            f.writefln(
+                "\"%s\",\"%s\",\"%s\",\"%s\",\"%s\",\"%s\",\"%s\",\"%s\",\"%s\",\"%s\",\"%s\"",
+                this.funcname, Clock.currTime.toISOExtString(), this.timeScale,
+                   this.rounds, this.wasInterrupted, this.avgStopWatch,
+                q0 > this.avgStopWatch ? q0 - this.avgStopWatch : 0,
+                q25 > this.avgStopWatch ? q25 - this.avgStopWatch : 0,
+                q50 > this.avgStopWatch ? q50 - this.avgStopWatch : 0,
+                q75 > this.avgStopWatch ? q75 - this.avgStopWatch : 0,
+                q100 > this.avgStopWatch ? q100 - this.avgStopWatch : 0
+            );
+        }
+    }
+}
+
+struct GenNum(T,T low, T high)
```

Comment:
space after comma - particularly jarring here where there's both space and no space after comma